lib/daru/maths/statistics/vector.rb

```diff
@@ -82,24 +82,38 @@ def factors
         #
         #   dv.max(2) { |a,b| a.size <=> b.size }
         #   #=> ["Jon Starkgaryen","Daenerys"]
+        def max(size=nil, &block)
+          data           = reject_values(*Daru::MISSING_VALUES).to_a
+          range          = size.nil? ? 0 : (0..size-1)
+          default_method = RUBY_VERSION >= '2.2'
+          default_method ? data.max(size, &block) : data.sort(&block).reverse[range]
+        end
```

Comment:
I have a LOT of questions to this solution :) I am generally unpleasant reviewer, you know.

First. Different implementations for a different Ruby versions is better to implement this way:
```ruby
if RUBY_VERSION >= '2.2'
  def implementation_method
  end
else
  def implementation_method
  end
end
```
Explanation: In your current version, `if` is performed every time `max` is called; in proposed one, it is performed only once, on files loading. And you'll see two different implementations would be cleaner, as only one of them needs `range`

Reply:
Good one.